tests/functional/settings_clients.js

```diff
@@ -178,9 +181,33 @@ define([
         .end()
 
         // clicking disconnect on the second device should update the list
-        .findByCssSelector('.client:nth-child(2) .client-disconnect')
-          .click()
-        .end()
+        .then(click('.client:nth-child(2) .client-disconnect'))
+
+        // get the modal dialog
+        .then(testElementExists('.intro'))
+        .then(testElementExists('.disabled'))
+
+        // test cancel
+        .then(click('.cancel-disconnect'))
+        .then(FunctionalHelpers.pollUntil(function () {
```

Comment:
Even if `waitForDeletedByCssSelector` doesn't work, I see this pattern repeated several times, it's probably worth creating a helper function in lib/helpers.js and using that.